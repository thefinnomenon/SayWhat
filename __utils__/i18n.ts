import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  ns: ['translations'],
  defaultNS: 'translations',
  resources: {},
  debug: false,
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
