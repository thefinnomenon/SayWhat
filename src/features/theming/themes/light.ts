const light: Theme = {
  nav: {
    statusBar: 'dark-content',
    background: '#f8f8f8',
    tintColor: '#000',
    activeTintColor: '#000',
    inactiveTintColor: '#a9a9a9',
    borderColor: '#e0e0e0',
    borderWidth: 1,
    titleStyle: {
      fontSize: 18,
      fontWeight: 'bold',
    },
  },
  colors: {
    background: '#fff',
    backgroundAlt: '#fff',
    border: '#e8e8e8',
    borderAlt: '#bdbdbd',
    primary: '#50b7ff',
    secondary: '#5c8bad',
    text: '#202020',
    textAlt: '#999',
  },
};

export default light;
