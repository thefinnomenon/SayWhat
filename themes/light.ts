import { Theme } from '../types';

const light: Theme = {
  theme: {
    nav: {
      background: '#f8f8f8',
      tintColor: '#000',
      activeTintColor: '#000',
      inactiveTintColor: '#a9a9a9',
      borderColor: '#e0e0e0',
      borderWidth: 1,
      titleStyle: {
        fontSize: 18,
        fontWeight: 'bold',
      },
    },
    colors: {
      background: '#fff',
      primary: '#50b7ff',
      secondary: '#5c8bad',
      border: '#e8e8e8',
      backgroundAlt: '#fff',
      borderAlt: '#bdbdbd',
      text: '#202020',
      textAlt: '#999',
    },
  },
};

export default light;
