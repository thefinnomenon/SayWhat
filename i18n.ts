import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as RNLocalize from 'react-native-localize';

const fallbackLanguage = { languageTag: 'sp', isRTL: false };
const defaultLanguage =
  RNLocalize.findBestAvailableLanguage(['en', 'pt']) || fallbackLanguage;

import en from './locales/en.json';
import pt from './locales/pt.json';

i18n.use(initReactI18next).init({
  lng: defaultLanguage.languageTag,
  resources: {
    en: { translation: en },
    pt: { translation: pt },
  },
  nsSeparator: false,
  keySeparator: false,
  fallbackLng: false,
  debug: false,
  interpolation: {
    escapeValue: false,
    formatSeparator: ',',
  },
  react: {
    wait: true,
  },
});

// RNLocalize.addEventListener('change', () => {
//   // do localization related stuffâ€¦
// });

export default i18n;
