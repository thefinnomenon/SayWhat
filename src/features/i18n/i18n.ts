import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as RNLocalize from 'react-native-localize';

// Define fallback language
const fallbackLanguage = { languageTag: 'en', isRTL: false };
// Set default language based upon system language
export const defaultLanguage =
  RNLocalize.findBestAvailableLanguage(['en', 'pt']) || fallbackLanguage;

// Load language files
import en from './locales/en.json';
import pt from './locales/pt.json';

// Initialize i18n module
i18n.use(initReactI18next).init({
  lng: defaultLanguage.languageTag,
  resources: {
    en: { translation: en },
    pt: { translation: pt },
  },
  nsSeparator: false,
  keySeparator: false,
  fallbackLng: false,
  debug: true,
  interpolation: {
    escapeValue: false,
    formatSeparator: ',',
  },
  react: {
    wait: true,
  },
});

export default i18n;
