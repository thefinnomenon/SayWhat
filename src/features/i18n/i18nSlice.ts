import { createSlice } from '@reduxjs/toolkit';
import { defaultLanguage } from './i18n';

const { actions, reducer } = createSlice({
  name: 'i18n',
  initialState: { currentLanguage: defaultLanguage.languageTag },
  reducers: {
    setLanguage(state, action) {
      state.currentLanguage = action.payload;
    },
    // REHYDRATE(state, action) {
    //   // i18n Set language
    // },
  },
});

export const { setLanguage } = actions;
export default reducer;
