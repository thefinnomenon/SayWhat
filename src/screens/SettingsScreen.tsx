import React from 'react';
import { ScrollView } from 'react-native';
import { Theme, ScreenProps } from '../../types';
import styled from 'styled-components/native';
import { useNavigation } from 'react-navigation-hooks';
import { useTranslation } from 'react-i18next';
import { useTheme } from '../../contexts/ManageThemeContext';
import i18n from 'i18next';
import Icon from 'react-native-vector-icons/FontAwesome5';
import Row from '../components/Row';

const languages = [
  { key: 'en', value: 'English' },
  { key: 'pt', value: 'Portuguese' },
];

const themes = [
  { key: 'system', value: 'System Default' },
  { key: 'light', value: 'Light' },
  { key: 'dark', value: 'Dark' },
];

export const SettingsScreen = () => {
  const theme = useTheme();
  const { t } = useTranslation();
  const { navigate } = useNavigation();

  return (
    <Container>
      <ScrollView>
        <Row
          icon={<StyledIcon name="globe-americas" />}
          title={t('Language')}
          onPress={() =>
            navigate('SettingList', {
              title: t('Language'),
              options: languages,
              selectedKey: i18n.language,
              onSelect: (language: string) => {
                if (language !== i18n.language) {
                  i18n.changeLanguage(language);
                }
              },
            })
          }
          renderAccessory={() => (
            <Value transform="uppercase">{i18n.language}</Value>
          )}
          showChevron
        />
        <Row
          icon={<StyledIcon name="eye" />}
          title={t('Theme')}
          onPress={() =>
            navigate('SettingList', {
              title: t('Theme'),
              options: themes,
              selectedKey: theme.mode,
              onSelect: (thm: string) => {
                theme.setMode(thm);
              },
            })
          }
          renderAccessory={() => (
            <Value transform="capitalize">{t(theme.mode)}</Value>
          )}
          showChevron
        />
      </ScrollView>
    </Container>
  );
};

SettingsScreen.navigationOptions = ({
  screenProps,
}: {
  screenProps: ScreenProps;
}) => ({
  title: screenProps.t('Settings'),
});

const Container = styled.View<Theme>`
  flex: 1;
  background: ${props => props.theme.colors.background};
`;

type ValueProps = {
  transform: 'uppercase' | 'lowercase' | 'capitalize' | 'none';
} & Theme;
const Value = styled.Text<ValueProps>`
  color: ${props => props.theme.colors.textAlt};
  text-transform: ${props => props.transform || 'none'};
  font-size: 18;
`;

const StyledIcon = styled(Icon)<Theme>`
  font-size: 20;
  margin-right: 12px;
  color: ${props => props.theme.colors.text};
`;

export default SettingsScreen;
