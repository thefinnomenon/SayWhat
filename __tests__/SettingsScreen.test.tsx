import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import light from '../src/features/theming/themes/light';
import { I18nextProvider } from 'react-i18next';
import i18n from './utilities/i18n';

import SettingsScreen from '../src/screens/SettingsScreen';

describe('The Settings Screen', () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store, wrapper: any;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = renderer.create(
      <Provider store={store}>
        <ThemeProvider theme={light}>
          <I18nextProvider i18n={i18n}>
            <SettingsScreen />
          </I18nextProvider>
        </ThemeProvider>
      </Provider>,
    );
  });

  it('renders correctly', () => {
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
