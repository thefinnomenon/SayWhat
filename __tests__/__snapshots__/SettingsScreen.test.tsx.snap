// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "height": 92,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "opacity": 1,
              "paddingLeft": 30,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#202020",
                      "fontSize": 40,
                      "marginRight": 24,
                    },
                  ],
                  Object {
                    "fontFamily": "FontAwesome5Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#202020",
                    "fontSize": 36,
                    "marginRight": 30,
                  },
                ]
              }
            >
              Language
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 24,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#999",
                    "fontSize": 36,
                    "textTransform": "uppercase",
                  },
                ]
              }
              transform="uppercase"
            >
              en-US
            </Text>
          </View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#202020",
                    "fontSize": 36,
                    "marginRight": 24,
                  },
                ],
                Object {
                  "fontFamily": "FontAwesome5Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "height": 92,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "opacity": 1,
              "paddingLeft": 30,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-start",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#202020",
                      "fontSize": 40,
                      "marginRight": 24,
                    },
                  ],
                  Object {
                    "fontFamily": "FontAwesome5Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#202020",
                    "fontSize": 36,
                    "marginRight": 30,
                  },
                ]
              }
            >
              Theme
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginRight": 24,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#999",
                    "fontSize": 36,
                    "textTransform": "capitalize",
                  },
                ]
              }
              transform="capitalize"
            >
              light
            </Text>
          </View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#202020",
                    "fontSize": 36,
                    "marginRight": 24,
                  },
                ],
                Object {
                  "fontFamily": "FontAwesome5Free-Solid",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "fontWeight": "900",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
